Set a `base` color and use it instead of `slate`

jagthedrummer opened the pull request:
Joint PR: https://github.com/bullet-train-co/bullet_train/pull/2267

Fixes: https://github.com/bullet-train-co/bullet_train-core/issues/1207

pascallaliberte reviewed (commented):
for super-select.css, there's a blue that's defined in select2.css that's not being overridden anywhere that we'll need to hunt down.

<img width="1480" height="642" alt="CleanShot 2025-10-29 at 18 23 02@2x" src="https://github.com/user-attachments/assets/2a656628-e5d9-42c5-b4fe-0295041e7865" />

pascallaliberte commented on bullet_train-themes-light/app/assets/stylesheets/light/fields/super_select.css:
```diff
@@ -109,7 +109,7 @@
 
   .select2-search__field {
```
Not sure which selector needs to be targeted yet, but there's a blue that's defined in select2.css that's not being overridden anywhere.

<img width="1480" height="642" alt="CleanShot 2025-10-29 at 18 23 02@2x" src="https://github.com/user-attachments/assets/2a656628-e5d9-42c5-b4fe-0295041e7865" />

jagthedrummer commented on bullet_train-themes-light/app/assets/stylesheets/light/fields/trix_editor.css:
```diff
@@ -87,13 +87,13 @@ trix-editor [data-trix-cursor-target] {
 trix-editor {
```
OK, this is weird. If I do the `tailwind.config.js` hack to change the `base` color, *then* I do get the lingering blue color. Somehow setting the `primary` and `secondary` color in `tailwind.config.js` does something slightly different than setting them in `theme.rb`. 😕

<img width="634" height="251" alt="CleanShot 2025-10-30 at 09 28 23" src="https://github.com/user-attachments/assets/0095d6b7-38ec-4067-b327-e6e9b99efdd4" />

jagthedrummer commented:
OK @pascallaliberte, I think this is ready for another review. I did find one hard-coded blue color that I changed to use an `@apply` directive. And I added some docs about changing `base` and the fact that you'll want `theme.rb` to agree with `tailwind.config.js`.
